### ISSUE TITLE
webMAN-MOD not ready issue

### ISSUE THREAD

**Issue by bodkAA** [aldostools/webMAN-MOD#320]
Hi. I have PS3 on rebug 4.85 lite. I've installed webMAN-MOD 1.47.25.9. When I try to got to webMAN settings or lauch a game, it says webMAN not ready. I've googled a lot about this. Nothing helped. It randomly works. After cold boot mostly it stops working. I do restart and it may work or may not. Can you suggest anything please? Thank you

**Comment by bodkAA** [comment]
I've done more experiments with this, and noticed it happens when I have external HD plugged in. When I use webMAN without it, everything works just fine. I'll try to check external drive for issues using standart Windows utilities and also will try to do defragmentation of external.

**Comment by bodkAA** [comment]
That didn't help. Checked few more times. Without external PS3 loads fast and webMAN works fine. When load PS3 with external webMAN works randomly fine. I noticed that with external PS3 loads sometimes fast(and webMAN works), sometimes for around 5-10 seconds longer and webMAN is not working(says not ready). And this is just random. (everything I'm doing is with disabled content scan at boot up). So this issue just doesn't make sense to me. Just random.

**Comment by Osama-Alhumaimidi** [comment]
I have noticed that webman mod is conflicting with the rebug xmb settings, try and disable the rebug xmb settings from the rebug settings app, restart and webman will start working properly

**Comment by bodkAA** [comment]
Are you talking about "Toggle XMB CFW setting" in rebug toolbox?

**Comment by Osama-Alhumaimidi** [comment]
yes, toggle that off (disable it so that the cfw menu doesnt appear in the network column in the xmb) and restart, now webman mod should work properly

**Comment by bodkAA** [comment]
I've seen that recommendation somewhere else, and did that before. Unfortunately didn't help. I've just enable it, rebooted ps3, disabled it again, done reboot again. Still says not ready. When go to webman setting get error(8071053d). When lauch game says webman not ready. Again happens randomly

**Comment by Osama-Alhumaimidi** [comment]
I just tried it multiple times,

1- enabled rebug XMB cfw settings
2- did a full restart through the xmb cfw settings
3- webman gone web"mad" and refused to start
4- to be sure, I turned the PS3 off for 5 mins, turned it on, the same issue happens
5- disabled the xmb cfw settings and restarted the console
6- loaded webman again (doesnt matter if full or simple)
7- now webman works correctly, to check, I turned off the console and let it rest for 5 mins, did a few restarts, and webman is still working

Edit:
I take it back, webman now refuses to start even with xmb cfw settings disabled

Edit 2:
webman mod must be reinstalled and started again in order to make it work (if you re-enabled then re-disabled the xmb cfw settings)

**Comment by bodkAA** [comment]
Anyway, really thank you for trying to help me. Issue still not fixed:(

**Comment by bodkAA** [comment]
Okay. I guess I've fixed the issue just by installing prepNTFS.pkg by pressing X(not holding x). I have external 1tb(700-fat32/300-ntfs). Tried to do cold boot many times with external and now everything seems to work fine. I'll try to use PS3 in a way it is now for few more days and if all good, I'll close the issue.

**Comment by Osama-Alhumaimidi** [comment]
the thing is I cannot find anything regarding this issue, its really hard to debug it this way

**Comment by bodkAA** [comment]
Something fails when PS3 starts with external. Cause as I said everything works fine without it. (External 1tb 700-fat32/300-ntfs). Tried installing prepntfs, thought maybe that will help, but no.

**Comment by bodkAA** [comment]
Since I didn't find any solution, and cause of this problem I was going crazy, I'll leave this here, cause I've fixed it only with this method. 
I doungraded my PS3 to Rebug 4.84, replaced my HDD in PS3, and installed Rebug 4.84 again and latest WebMAN Mod. I've tried to do full formatting on my old HDD so that didn't help(and there is no problem with HDD itself, I've checked it with Victoria, so it's fine). It seems to me that PS3 remembers last HDD and there were some issues in flash memory or smth(I'm not sure), but just replacing HDD helped(it probably clears everything on PS3 itself). Console works fine for the last week with and without plugged in external. So maybe this will help you to find some solution or easier fix on the problem @aldostools. Thank you anyway

**Comment by Osama-Alhumaimidi** [comment]
for me unless there is a USB storage connected, I keep getting the webmanMod not ready issue, its like its stuck looking for a USB device on boot even when I disable content scan. Its not an incompatibility with rebug tools, because I just enabled all rebug cfw settings and webmanMOD works fine as long as there is a USB storage connected, if I disconnect and restart the console, it goes crazy again

**Comment by Osama-Alhumaimidi** [comment]
ok I have an update, I think I just discovered the cause of the issue, setting the fan speed to SYSCON will break webmanMOD, set it back t dynamic and webmanMOD will work again

**Comment by bodkAA** [comment]
After I've changed hdd, installed rebug, installed latest webman, I anyway set syscon and as i said it works fine now.(i had that issue even with dynamic fan set up +it was to loud for me) I just assume it's not a source of the issue in my case(but of course i can be wrong).
I also know that some people had webman mod not ready issue with enabled content scan at boot up and disabling helped them, if u c what I'm saying.. but anyway thank you

**Comment by Osama-Alhumaimidi** [comment]
also if you set fan speed to SYSCON, the only way to get webmanMOD to work is to connect a USB storage device, which is weird because the two should be unrelated

**Comment by aldostools** [comment]
FYI the message "webMAN not ready" is produced by wm_proxy.sprx

The wm_proxy.sprx is an interface between the XMB and webMAN. 
It the user selection from XMB and calls the corresponding web command via HTTP.

The error "webMAN not ready" is displayed when the wm_proxy cannot connect to local port 80, which is the web server port where webMAN receives the remote requests using the HTTP protocol.

There are various reasons why this happen, then most common are:
1- webMAN MOD is not running. Check if the file boot_plugin.txt exists and the path inside the file is the correct path of webMAN MOD. e.g. /dev_hdd0/plugins/webftp_server.sprx

The boot_plugin.txt is removed by Cobra after a Rebuild Database from Recovery.

2- webMAN MOD may be busy. The plugin accept a limited number of requests. If all the slots are in used, webMAN will reject the connection.

3- The network is disabled or port 80 is already in use..

4- The plugin was unloaded or could not be loaded correctly or there is not enough memory to load the plugin.

**Comment by bodkAA** [comment]
Before I did HDD replacement, I formatted my PS3 with old hdd completely(using acronis on pc), then just installed webman mod and still had that random webman not ready issue(only with external drive plugged in, without it everything was working fine). Only thing that helped was HDD replacement.(i just flipped my hdd from laptop to ps3 and from ps3 to laptop). Ps3 is working fine with no issue now(with or without same external).
@aldostools All steps that you mentioned before were checked, in my case that didn't help. Thank you again

**Comment by aldostools** [comment]
It looks like the HDD had some bad sectors. I'm glad that it's working again with no issues.

**Comment by bodkAA** [comment]
As I mentioned before, hdd that is now in laptop(it was in ps3 before, when i had this issue with webman) was checked on bad sectors and has none. It's fine. It seems to me that ps3 remembers somehow last hdd in flash memory, and when you change it, everything clears(maybe even some errors) I guess so, but I'm not sure. If there would be bad sectors, I'd say that it could've been the source of issue, but no, hdd was fine(checked using Victoria utility).

**Comment by Joonie86** [comment]
No further feedback
